IdentityCore/src/controllers/MSIDSilentController.m

```diff
@@ -121,28 +122,48 @@ - (void)acquireTokenWithRequest:(MSIDSilentTokenRequest *)request
             MSIDTelemetryAPIEvent *telemetryEvent = [self telemetryAPIEvent];
             [telemetryEvent setUserInformation:result.account];
             [telemetryEvent setIsExtendedLifeTimeToken:result.extendedLifeTimeToken ? MSID_TELEMETRY_VALUE_YES : MSID_TELEMETRY_VALUE_NO];
+            if (self.isLocalFallbackMode)
+            {
+                 [telemetryEvent setSsoExtFallBackFlow:1];
+            }
+            
             [self stopTelemetryEvent:telemetryEvent error:error];
 #endif
             self.currentRequest = nil;
             
             completionBlock(result, error);
             return;
         }
-
+        
         self.currentRequest = nil;
-        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *ssoResult, NSError *ssoError)
+        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *fallbackResult, NSError *fallbackError)
         {
             // We don't have any meaningful information from fallback controller (edge case of SSO error) so we use the local controller result earlier
-            if (!ssoResult && (ssoError.code == MSIDErrorSSOExtensionUnexpectedError))
+            // In if block, we handle fallback from local controller to Sso Extention controller, and to use the valid error information
+            if (self.controllerType == MSIDSSOExtensionSilentTokenRequestControllerType)
             {
-                completionBlock(result, error);
+                if (fallbackResult)
+                {
+                    completionBlock(fallbackResult, nil);
+                }
+                else
+                {
+                    completionBlock(nil, (error.code == MSIDErrorSSOExtensionUnexpectedError) ? fallbackError : error);
```

Comment:
I'm still not sure why this is needed. If SSO extension returns MSIDErrorSSOExtensionUnexpectedError, then logic should fallback to the next controller (silent), and we can return errors from that controller.

Reply:
I got the idea, but if added something like error.code == MSIDErrorSSOExtensionUnexpectedError, continue with fallback controller in the abstract layer, it will break existing flow. Current flow is that as long as there is fallback controller, will use it